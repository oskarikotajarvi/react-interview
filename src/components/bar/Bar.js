import React, { useState } from 'react';
import './bar.css';

const Bar = ({ onSubmit }) => {
  const [newTodo, setNewTodo] = useState('');

  //On form submit
  const submit = e => {
    e.preventDefault();
    onSubmit(newTodo);
    setNewTodo('');
  };

  return (
    <div className="wrapper">
      <form className="bar-form" onSubmit={submit} data-testid="bar-form">
        <div>
          <input
            className="bar-form-control"
            type="text"
            name="newTodo"
            value={newTodo}
            onChange={({ target }) => setNewTodo(target.value)}
            required
            data-testid="bar-input"
          />
          <label
            className="bar-form-control-placeholder"
            htmlFor="newTodo"
            data-testid="bar-label"
          >
            Add new todo
          </label>
          <button
            className="btn btn-submit"
            type="submit"
            value="Submit"
            data-testid="bar-submit"
          >
            Submit
          </button>
        </div>
      </form>
    </div>
  );
};

export default Bar;
