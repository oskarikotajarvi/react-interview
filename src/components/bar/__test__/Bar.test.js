import React from 'react';
import ReactDOM from 'react-dom';
import {
  render,
  cleanup,
  fireEvent,
  getByTestId
} from '@testing-library/react';
import renderer from 'react-test-renderer';
import '@testing-library/jest-dom/extend-expect';
import Bar from '../Bar';

afterEach(cleanup);

describe('Bar tests...', () => {
  it('Renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(
      <Bar
        onSubmit={() => {
          return;
        }}
      />,
      div
    );
  });

  it('Renders the form correctly', () => {
    const { container } = render(
      <Bar
        onSubmit={() => {
          return;
        }}
      />
    );
    const input = getByTestId(container, 'bar-input');
    const label = getByTestId(container, 'bar-label');
    const submit = getByTestId(container, 'bar-submit');
    expect(input).toHaveTextContent('');
    expect(label).toHaveTextContent('Add new todo');
    expect(submit).toHaveTextContent('Submit');
    expect(input.value).toBe('');
  });

  it('Inputting a value to the input field changes input value', () => {
    const { container } = render(
      <Bar
        onSubmit={() => {
          return;
        }}
      />
    );
    const input = getByTestId(container, 'bar-input');
    const newTodoValue = 'Get the laundry';

    expect(input.value).toBe('');
    fireEvent.change(input, { target: { value: newTodoValue } });
    expect(input.value).toBe(newTodoValue);
  });

  it('Submit button fires', () => {
    const { container } = render(
      <Bar
        onSubmit={() => {
          return;
        }}
      />
    );
    const input = getByTestId(container, 'bar-input');
    const submit = getByTestId(container, 'bar-submit');

    fireEvent.change(input, { target: { value: 'new Todo' } });
    expect(input.value).toBe('new Todo');
    fireEvent.click(submit);
    expect(input.value).toBe('');
  });

  it('Matches snapshot', () => {
    const tree = renderer
      .create(
        <Bar
          onSubmit={() => {
            return;
          }}
        />
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
