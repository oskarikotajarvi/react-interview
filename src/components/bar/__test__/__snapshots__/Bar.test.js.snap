// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Bar tests... Matches snapshot 1`] = `
<div
  className="wrapper"
>
  <form
    className="bar-form"
    data-testid="bar-form"
    onSubmit={[Function]}
  >
    <div>
      <input
        className="bar-form-control"
        data-testid="bar-input"
        name="newTodo"
        onChange={[Function]}
        required={true}
        type="text"
        value=""
      />
      <label
        className="bar-form-control-placeholder"
        data-testid="bar-label"
        htmlFor="newTodo"
      >
        Add new todo
      </label>
      <button
        className="btn btn-submit"
        data-testid="bar-submit"
        type="submit"
        value="Submit"
      >
        Submit
      </button>
    </div>
  </form>
</div>
`;
